iup/__init__.py

```diff
@@ -153,69 +153,63 @@ def to_incident(self, group_cols: tuple[str,] | None) -> IncidentUptakeData:
 
         return IncidentUptakeData(out)
 
+    def insert_rollout(
+        self, rollout: List[dt.date], group_cols: tuple | None
+    ) -> pl.DataFrame:
+        """
+        Insert into cumulative uptake data rows with 0 uptake on rollout dates.
 
-def insert_rollout(
-    frame: pl.DataFrame, rollout: dt.date, group_cols: dict | None
-) -> pl.DataFrame:
-    """
-    Insert into NIS uptake data rows with 0 uptake on the rollout date.
+        Parameters
+        rollout: List[dt.date]
+            list of rollout dates
+        group_cols: tuple[str] | None
+            names of grouping factor columns
 
-    Parameters
-    frame: pl.DataFrame
-        NIS data in the midst of parsing
-    rollout: dt.date
-        rollout date
-    group_cols: dict | None
-        dictionary of the NIS columns for the grouping factors
-        keys are the NIS column names and values are the desired column names
+        Returns
+            cumulative uptake data with rollout rows included
 
-    Returns
-        NIS cumulative data with rollout rows included
+        Details
+        A separate rollout row is added for every grouping factor combination.
+        """
+        frame = self
 
-    Details
-    A separate rollout row is added for every grouping factor combination.
-    """
-    if group_cols is not None:
-        rollout_rows = (
-            frame.select(pl.col(v) for v in group_cols.values())
-            .unique()
-            .with_columns(date=rollout, estimate=0.0)
-        )
-    else:
-        rollout_rows = pl.DataFrame({"date": rollout, "estimate": 0.0})
+        for r in rollout:
+            if group_cols is not None:
+                rollout_rows = (
+                    frame.select(pl.col(v) for v in group_cols)
```

Comment:
```suggestion
                    frame.select(group_cols)
```
you can pass a list of strings to `.select()`, conveniently, so long as you're not manipulating the columns at the same time

Reply:
Also nice!